src/expenses/services/expense.service.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const helper = require('../../helper');
+
+let expenses = [];
+
+const clearState = () => {
+  expenses = [];
+};
+
+const getAll = ({ userId, categories, from, to }) => {
+  let newExpences = [...expenses];
+
+  if (userId) {
+    newExpences = newExpences.filter(
+      expence => expence.userId === Number(userId)
+    );
+  }
+
+  if (categories) {
+    const categoriesList = Array.isArray(categories)
+      ? categories
+      : [categories];
+
+    newExpences = newExpences.filter(
+      expense => categoriesList
+        .includes(expense.category)
+    );
+  }
+
+  if (from) {
+    newExpences = newExpences.filter(
+      expense => new Date(from) <= expense.spentAt
+    );
+  }
+
+  if (to) {
+    newExpences = newExpences.filter(
+      expense => new Date(to) >= expense.spentAt
+    );
+  }
+
+  return newExpences;
+};
+
+const getById = (id) => {
+  return expenses.find(expense => expense.id === id) || null;
+};
+
+const create = (body) => {
+  const expense = {
+    id: helper.getId(expenses),
+    userId: Number(body.userId),
+    spentAt: body.spentAt ? new Date(body.spentAt) : undefined,
+    title: body.title,
+    amount: Number(body.amount),
+    category: body.category,
+    note: body.note,
```

Comment:
Maybe better to check if types are correct and return error if not, than to transform them